src/partr.c

```diff
@@ -424,6 +424,11 @@ JL_DLLEXPORT jl_task_t *jl_task_get_next(jl_value_t *trypoptask, jl_value_t *q,
                     continue;
                 }
             }
+            else if (ptls->tid != 0 && !jl_atomic_load_relaxed(&_threadedregion)) {
+                // Since we might have started some IO work, we might need to ensure
+                // tid = 0 will go watch that new event source for us.
+                (void)wake_thread(0);
```

Comment:
This does not wake thread 0 quite properly enough to merge this yet. I need to work more to figure out if we need another sleep_check_state (to distinguish sleeping in uv_run vs entirely stopped), or whether to take some other approach towards checking here such as:
```diff
diff --git a/src/partr.c b/src/partr.c
index 4faff409c7..3ab2cebe0d 100644
--- a/src/partr.c
+++ b/src/partr.c
@@ -425,6 +425,14 @@ JL_DLLEXPORT jl_task_t *jl_task_get_next(jl_value_t *trypoptask, jl_value_t *q,
                 }
             }
 
+            if (ptls->tid != 0 && !jl_atomic_load_relaxed(&_threadedregion)) {
+                // Since we might have started some IO work, we might need to ensure
+                // tid = 0 will go watch that new event source for us.
+                jl_task_t *uvlock = jl_atomic_load_relaxed(&jl_uv_mutex.owner);
+                if (uvlook == NULL)
+                    jl_wakeup_thread(0);
+            }
+
             // the other threads will just wait for an individual wake signal to resume
             JULIA_DEBUG_SLEEPWAKE( ptls->sleep_enter = cycleclock() );
             int8_t gc_state = jl_gc_safe_enter(ptls);
```